pyaedt/modules/MeshIcepak.py

```diff
@@ -1,12 +1,912 @@
+        "MaxElementSizeY": "0.02mm",
+        "MaxElementSizeZ": "0.03mm",
+        "MinElementsInGap": "3",
+        "MinElementsOnEdge": "2",
+        "MaxSizeRatio": "2",
+        "NoOGrids": False,
+        "EnableMLM": True,
+        "EnforeMLMType": "3D",
+        "MaxLevels": "0",
+        "BufferLayers": "0",
+        "UniformMeshParametersType": "Average",
+        "2DMLMType": "2DMLM_None",
+        "MinGapX": "1mm",
+        "MinGapY": "1mm",
+        "MinGapZ": "1mm",
+    }
+    aedt_20212_args = [
+        "ProximitySizeFunction",
+        "CurvatureSizeFunction",
+        "EnableTransition",
+        "OptimizePCBMesh",
+        "Enable2DCutCell",
+        "EnforceCutCellMeshing",
+        "Enforce2dot5DCutCell",
+    ]
+
+    def __init__(self, mesh_class, app):
+        self._app = app
+        self._mesh_class = mesh_class
+        self.instance_settings = self.common_mesh_settings.copy()
+        self.instance_settings.update(self.manual_mesh_settings.copy())
+        self.instance_settings.update(self.automatic_mesh_settings.copy())
+        if settings.aedt_version < "2021.2":
+            for arg in self.aedt_20212_args:
+                del self.instance_settings[arg]
+
+    @pyaedt_function_handler()
+    def _dim_arg(self, value):
+        if isinstance(value, str):
+            return value
+        else:
+            return _dim_arg(value, getattr(self._mesh_class, "_model_units"))
+
+    def parse_settings(self):
+        """
+        Parse mesh region settings.
+
+        Returns
+        -------
+        dict
+            List of strings containing all the parts that must be included in the subregion.
+        """
+        out = []
+        for k, v in self.instance_settings.items():
+            out.append(k + ":=")
+            if k in ["MaxElementSizeX", "MaxElementSizeY", "MaxElementSizeZ", "MinGapX", "MinGapY", "MinGapZ"]:
+                v = self._dim_arg(v)
+            out.append(v)
+        return out
+
+    def _key_in_dict(self, key):
+        if self._mesh_class.manual_settings:
+            ref_dict = self.manual_mesh_settings
+        else:
+            ref_dict = self.automatic_mesh_settings
+        if key in ref_dict or key in self.common_mesh_settings:
+            return True
+        else:
+            return False
+
+    def __getitem__(self, key):
+        if key == "Level":
+            key = "MeshRegionResolution"
+        if self._key_in_dict(key):
+            return self.instance_settings[key]
+        else:
+            raise KeyError("Setting not available.")
+
+    def __setitem__(self, key, value):
+        if key == "Level":
+            key = "MeshRegionResolution"
+        if self._key_in_dict(key):
+            if key == "MeshRegionResolution":
+                try:
+                    value = int(value)
+                    if value < 1:
+                        self._app.logger.warning("Minimum resolution value is 1.")
+                        value = 1
```

Comment:
Not sure if it's a good idea to change the user value like that.
I would rather not doing anything than changing the value as it could confuse the user.

Reply:
Should I raise an exception?

Reply:
IMO it would be for the best but again, we should try to keep consistency for the moment.
@Samuelopez-ansys How would you handle this case ?

Reply:
I would say that returning False does not make sense, as this is a setter. Either I raise an exception or I log a pyaedt error and do nothing.

Reply:
I prefer to use, at the moment, @lorenzovecchietti approach.

Reply:
Ok, so I updated the log messages and now it is clearly stated that we are overriding the input